src/main/java/io/jenkins/infra/repository_permissions_updater/run/PermissionJson.java

```diff
@@ -0,0 +1,17 @@
+package io.jenkins.infra.repository_permissions_updater.run;
+
+import java.util.List;
+import java.util.Map;
+
+public record PermissionJson(
+        String name,
+        String includesPattern,
+        String excludesPattern,
+        List<String> repositories,
+        Principals principals
+) {
+    public record Principals(
+            Map<String, List<String>> users,
+            Map<String, List<String>> groups
+    ) {}
+}
```

Comment:
```suggestion
}

```

nit: posix text file should always end with an empty endline. Ref. https://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap03.html#tag_03_392
Worth setting up your IDE to automatically add it when saving